### PR TITLE
Prepare for PHP7.3, 

### PR DESCRIPTION
**Pull request by Dwarfex**
Hi, 
yes i know this PR is a bit long. I put each type of change in a separate commit to make it easier viewable.

TLDR: I fixed some potential bugs, replaced (soon to bee) deprecated functions 

After i run phpstan, phan and PHPStorm Inspections, there was a huge list and i started to fix some(in my perspective) major ones.

- Security: allow only de-serialization to ProductList::class
- fix bug where "get_party_dropdown_form" would always show old (disregarding $show_old parameter)
- fix possible bug where LanSuite\Module\CashMgr\Account::class would disregard $party_id
- removed identical, duplicated array entries
- added used, but not mentioned extensions to composer file
- remove setting magic quotes to "0" as its always off (doesn't exist anymore)
- replace deprecated 'while = each()' in favor for 'foreach'
- replace eregi() with preg_match() (case insensitive)
- replace function calls with inbuilt constants
- replace alias function calls with originals
- performance (x2): replace 'array_push($array, $value)' with '$array[] = $value';
- replace 'for' loops with object counting in favor for 'foreach'
- clarified parentheses(added brackets for "how php does interpret"): needs validation if this is actual wanted behavior
- joined switch statements that do exactly the same for better overview

Let me know if there are any changes - i'm happy to rework if needed - so this can be merged!
P.S.: Yes - i changed files in /ext_scripts on purpose as i don't know what is really used there. (thats some work for the future i'd guess ;) ) 

Greetings,
Dwarfex

### CODE REVIEW

**Comment by Dwarfex** [comment]
currently fixing Jenkins Code style markings

**Comment by andygrunwald** [comment]
@Dwarfex Great work! Thank you! I will need a bit of time to review this.

>  currently fixing Jenkins Code style markings
As you mentioned already, this PR is rather big.
As long as the failures are the only style related, I suggest to ignore them + fix them in a new PR.
By this, we avoid that this PR is getting bigger + we are able to differ functionality changes with style changes.

Feel free to open a new PR for the style changes and write that the new PR depends on this one and should not be merged earlier.

What do you think?

**Comment by Dwarfex** [comment]
yeah - in the curent state the PR is 'big' but rather 'clean'

I agree with you on opening a seperate PR.

Maybe it would be best to run phpcs and phpcbf in a seperate branch to cleanup the whole master branch once.
So one could prevent this issue from happening :)

File: ext_scripts/mobile_device_detect.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ function mobile_device_detect($iphone=true,$ipad=true,$android=true,$opera=true,
...
...apt'=>'wapt','winc'=>'winc','winw'=>'winw','wonu'=>'wonu','x700'=>'x700','xda2'=>'xda2','xdag'=>'xdag','yas-'=>'yas-','your'=>'your','zte-'=>'zte-','zeto'=>'zeto','aste'=>'aste','audi'=>'audi','avan'=>'avan','blaz'=>'blaz','brew'=>'brew','brvw'=>'brvw','bumb'=>'bumb','ccwa'=>'ccwa','cell'=>'cell','cldc'=>'cldc','cmd-'=>'cmd-','dang'=>'dang','eml2'=>'eml2','fetc'=>'fetc','hipt'=>'hipt','http'=>'http','ibro'=>'ibro','idea'=>'idea','ikom'=>'ikom','ipaq'=>'ipaq','jbro'=>'jbro','jemu'=>'jemu','jigs'=>'jigs','keji'=>'keji','kyoc'=>'kyoc','kyok'=>'kyok','libw'=>'libw','m-cr'=>'m-cr','midp'=>'midp','mmef'=>'mmef','moto'=>'moto','mwbp'=>'mwbp','mywa'=>'mywa','newt'=>'newt','nok6'=>'nok6','o2im'=>'o2im','pant'=>'pant','pdxg'=>'pdxg','play'=>'play','pluc'=>'pluc','port'=>'port','prox'=>'prox','rozo'=>'rozo','sama'=>'sama','seri'=>'seri','smal'=>'smal','symb'=>'symb','tosh'=>'tosh','treo'=>'treo','upsi'=>'upsi','vx52'=>'vx52','vx53'=>'vx53','vx60'=>'vx60','vx61'=>'vx61','vx70'=>'vx70','vx80'=>'vx80','vx81'=>'vx81','vx83'=>'vx83','vx85'=>'vx85','wap-'=>'wap-','webc'=>'webc','whit'=>'whit','wmlb'=>'wmlb','xda-'=>'xda-',))); // check against a list of trimmed user agents to see if we find a match
```
Comment (andygrunwald) [inline ext_scripts/mobile_device_detect.php:147]: It seems that in the new version are fewer entries.
@Dwarfex Do you know what happened here?

File: modules/foodcenter/statchange.php [hunk 2]
```diff
@@ -152,7 +152,8 @@
         $handle = opendir("ext_inc/foodcenter_templates");
         while ($file = readdir($handle)) {
             if (($file != ".") and ($file != "..") and ($file != ".svn") and (!is_dir($file))) {
-                if ((substr($file, -3, 3) == "htm") && (substr($file, -7, 7) != "row.htm") || (substr($file, -4, 4) == "html") && (substr($file, -8, 8) != "row.html")) {
+                //@TODO: validate this parentheses:
```
Comment (andygrunwald) [inline modules/foodcenter/statchange.php:155]: parentheses looks correct. Please remove this TODO

File: modules/paypal/Classes/PayPal.php [hunk 3]
```diff
@@ -135,7 +135,8 @@ public function createPaymentLink()
 
             // use the same link as the user currently has to avoid to return to a variant where the user is not logged in
             // e.g. HTTP vs. HTTPS or http://www.something vs. http://something.de
-            if (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) {
+            //@TODO: validate this parentheses:
```
Comment (andygrunwald) [inline modules/paypal/Classes/PayPal.php:138]: parentheses looks correct. Please remove TODO

File: modules/usrmgr/Classes/UserManager.php [hunk 4]
```diff
@@ -37,7 +37,8 @@ public function SendVerificationEmail($id)
             }
             $verification_link .= "index.php?mod=usrmgr&action=verify_email&verification_code=$verification_code";
         } else {//No HTTPS URL defined, but maybe the user is already using it?
-            if (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) {
+            //@TODO: validate this parentheses:
```
Comment (andygrunwald) [inline modules/usrmgr/Classes/UserManager.php:40]: parentheses looks correct. Please remove todo

File: modules/usrmgr/remind.php [hunk 5]
```diff
@@ -30,7 +30,8 @@
                     } //make sure that it ends with a slash
                     $verification_link .= "index.php?mod=usrmgr&action=pwrecover&step=3&fcode=$fcode";
                 } else { // fallback to old version
-                    if (!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443) {
+                    //@TODO: validate this parentheses:
```
Comment (andygrunwald) [inline modules/usrmgr/remind.php:33]: parentheses looks correct. Please remove todo

**Review by andygrunwald — CHANGES_REQUESTED** [review]
Overall it looks really good. Thanks a lot @Dwarfex!
I left a few smaller comments. Looking forward to merge this.

File: ext_scripts/mobile_device_detect.php (hunk 1 above)
Comment (Dwarfex) [reply ext_scripts/mobile_device_detect.php:147]: yes - here were duplicated entries, that i removed

**Comment by Dwarfex** [comment]
@andygrunwald  i removed the todos :)

**Review by andygrunwald — APPROVED** [review]

**Comment by andygrunwald** [comment]
Thanks a lot :)
